qa/integration-tests/src/test/java/io/camunda/it/client/QueryTest.java

```diff
@@ -111,7 +111,7 @@ public static void waitForFlowNodeInstances(
   public static void waitForProcessesToBeDeployed(
       final CamundaClient camundaClient, final int expectedProcessDefinitions) {
     Awaitility.await("should deploy processes and import in Operate")
-        .atMost(Duration.ofSeconds(15))
+        .atMost(Duration.ofSeconds(120))
```

Comment:
🙃 :x: I feel like this is a suboptimal approach and quite fragile.

* In general I think we should try to have a minimal timeout.
* I do understand that we can't have this for AWS as we use small instances, etc. But there is a problem with this approach, new tests were likely not to be written with this in mind. This means engineers will put again, random values and likely too small ones on new tests.
* How can we make sure that new tests are not failing, because of this? How can we make sure that people are not reducing this again, because they are wondering why the defaults are so high?


@npepinpe do you have anything to add here or maybe some idea?

Reply:
We might want to have some constants, and better documentation regarding the reasons.

Maybe as you [proposed](https://github.com/camunda/camunda/pull/27828/files#r1963123281) we can make it somehow configurable from the CI, that only the AWS pipeline has such big timeouts.

Reply:
One major issue is that different tests use different time outs and I don't know whether those timeout are taken out of thin air or that actually something required. I see few 15 seconds timeouts and few 60 seconds. This may mean that we might need two separate constants which is not great because whenever we have 2, there will be 3-4-5 at some point.

With my next commit I will try to make all unified with 15 seconds for local and parametrized for CI and see if it helps.

Reply:
I will also ask tomorrow infra team to increase AWS instance types with more CPU/memory

Reply:
Instance type tickets: https://github.com/camunda/team-infrastructure/issues/787

Reply:
Hi @ChrisKujawa do you have other suggestions or can we get it moving? Maintaining this PR becomes a challenging task.